core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -524,6 +528,19 @@ class SparkContext(config: SparkConf) extends Logging {
     executorEnvs ++= _conf.getExecutorEnv
     executorEnvs("SPARK_USER") = sparkUser
 
+    val configuredPluginClasses = conf.get(SHUFFLE_IO_PLUGIN_CLASS)
+    val maybeIO = Utils.loadExtensions(
+      classOf[ShuffleDataIO], Seq(configuredPluginClasses), conf)
+    require(maybeIO.nonEmpty, s"At least one valid shuffle plugin must be specified by config " +
+      s"${SHUFFLE_IO_PLUGIN_CLASS.key}, but $configuredPluginClasses resulted in zero valid " +
+      s"plugins.")
+    require(maybeIO.size == 1,
+        s"Specified shuffle plugin(s) $configuredPluginClasses resulted in more than one valid " +
+        s"plugin, but only one valid plugin should be specified")
+    _shuffleDriverComponents = maybeIO.head.driver()
+    _shuffleDriverComponents.initializeApplication().asScala.foreach {
+      case (k, v) => _conf.set(ShuffleDataIO.SHUFFLE_SPARK_CONF_PREFIX + k, v) }
```

Comment:
Lets log these extra confs at least at the debug level -- though I think I'd prefer they were logged at info.

is the user alliowed to set "spark.shuffle.plugin." confs themselves?  Does the driver component always get to override whatever the user has set?

I don't have any particular reason to think it should be one way or another, just want to check if you have any reason to want it to be one way or the other.

Reply:
That's one use case I didn't think about. Intuitively, I guess it would make sense for the user to be able to override the configs set by the plugin, since the alternative might be confusing to a user. In that case, I think it would make sense to log the configs, and also log when the value returned by the driver won't take effect due to user overrides for that particular application.

Reply:
In general I'm wary of logging configurations, particularly if the plugin implementation can include configurations that may be sensitive. Since this isn't going through the existing paths of Spark configuration logging - which, when that is done, redacts configuration values accordingly - I'd err on the side of caution and not do logging here.